src/App.tsx

```diff
@@ -1,14 +1,35 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [todoId, setTodoId] = useState(0);
+  const [status, setStatus] = useState('All');
+
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then((todoList) => {
+        setTodos(todoList);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code